pylsp/workspace.py

```diff
@@ -135,15 +136,43 @@ def report_progress(
         message: Optional[str] = None,
         percentage: Optional[int] = None,
     ) -> Generator[Callable[[str, Optional[int]], None], None, None]:
-        token = self._progress_begin(title, message, percentage)
+        if self._config:
+            client_supports_progress_reporting = (
+                self._config.capabilities.get("window", {}).get("workDoneProgress", False)
+            )
+        else:
+            client_supports_progress_reporting = False
+
+        if client_supports_progress_reporting:
+            try:
+                token = self._progress_begin(title, message, percentage)
+            except Exception:  # pylint: disable=broad-exception-caught
+                log.warning(
+                    "error while trying to initialize progress reporting."
+                    "Either the editor failed, was too slow to answer, or this LSP "
```

Comment:
There should never be a need for a timeout. Client is supposed to respond with an Error if it doesn't understand the request. But since you are checking the capability then that should not happen even.

Reply:
The second reason for the timeout is plugin developers. 

Assuming a plugin dev is using the progress reporting in a synchronous handler. 

Without the timeout this request would just hang, and the response for the create-token would only be received after the handler is finished. 

Without knowing more about internals of the lsp protocol, and this library, I didn't see an other way to prevent the request just hanging.

Reply:
Seeing your edit, what do you propose?

Reply:
OK, but then the next line of this warning is more relevant anyway?

Reply:
ah, so it's about the error message, right? 

I updated it, I hope better to understand